Fails to start gopls, "not a directory"

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. clone https://github.com/llimllib/git-ls
2. open `main.go`
3. note that there's no LSP, and a failure I don't understand is in the logs

The logs have the error: ` err: fork/exec /opt/homebrew/bin/go: not a directory` (see below for context). The error is correct in that /opt/homebrew/bin/go is not a directory, but a symlink to my go binary:

```
$ which go
/opt/homebrew/bin/go
$ ls -alh /opt/homebrew/bin/go
lrwxr-xr-x 1 llimllib 26 May  2 11:15 /opt/homebrew/bin/go -> ../Cellar/go/1.22.2/bin/go*
```

gopls works as expected in neovim, with the same go and gopls install:

```
 Client: gopls (id: 2, bufnr: [1])
 	filetypes:       go, gomod, gowork, gotmpl
 	autostart:       true
 	root directory:  /Users/llimllib/code/git-ls
 	cmd:             /Users/llimllib/go/bin/gopls
```

`~/go/bin/gopls` is version 0.15.2:

```
$ ~/go/bin/gopls version
golang.org/x/tools/gopls v0.15.2
```

### Environment

Zed: v0.133.7 (Zed)
OS: macOS 14.4.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

```
2024-05-03T11:54:01-04:00 [INFO] starting language servers for Go: gopls
2024-05-03T11:54:01-04:00 [INFO] starting language servers for JSON: json-language-server
2024-05-03T11:54:01-04:00 [INFO] stopping language server gopls
2024-05-03T11:54:01-04:00 [WARN] Generic lsp request to gopls failed: oneshot canceled
2024-05-03T11:54:01-04:00 [ERROR] crates/lsp/src/lsp.rs:768: oneshot canceled
2024-05-03T11:54:01-04:00 [WARN] Generic lsp request to gopls failed: oneshot canceled
2024-05-03T11:54:01-04:00 [ERROR] crates/lsp/src/lsp.rs:352: cannot read LSP message headers
2024-05-03T11:54:01-04:00 [ERROR] crates/project/src/project.rs:6949: oneshot canceled
2024-05-03T11:54:01-04:00 [INFO] starting language servers for Go: gopls
2024-05-03T11:54:01-04:00 [INFO] starting language server "gopls", path: "/Users/llimllib/code/git-ls/main.go", id: 4
2024-05-03T11:54:01-04:00 [ERROR] crates/editor/src/editor.rs:3876: oneshot canceled
2024-05-03T11:54:01-04:00 [INFO] found user-installed language server for Go. path: "/Users/llimllib/go/bin/gopls", arguments: ["-mode=stdio"]
2024-05-03T11:54:01-04:00 [INFO] starting language server. binary path: "/Users/llimllib/go/bin/gopls", working directory: "/Users/llimllib/code/git-ls", args: ["-mode=stdio"]
2024-05-03T11:54:01-04:00 [INFO] Language server with id 4 sent unhandled notification window/showMessage:
{
  "type": 1,
  "message": "Error loading workspace folders (expected 1, got 0)\nfailed to load view for file:///Users/llimllib/code/git-ls/main.go: err: fork/exec /opt/homebrew/bin/go: not a directory: stderr: \n"
}
2024-05-03T11:54:01-04:00 [WARN] Generic lsp request to node failed: Unhandled method textDocument/definition
2024-05-03T11:54:01-04:00 [INFO] Language server with id 4 sent unhandled notification client/unregisterCapability:
{
  "unregisterations": [
    {
      "id": "workspace/didChangeWatchedFiles-0",
      "method": "workspace/didChangeWatchedFiles"
    }
  ]
}
```

--- a/crates/lsp/src/lsp.rs
+++ b/crates/lsp/src/lsp.rs
@@ -79,6 +79,7 @@ pub struct LanguageServer {
     io_tasks: Mutex<Option<(Task<Option<()>>, Task<Option<()>>)>>,
     output_done_rx: Mutex<Option<barrier::Receiver>>,
     root_path: PathBuf,
+    working_dir: PathBuf,
     server: Arc<Mutex<Option<Child>>>,
 }
 
@@ -288,6 +289,7 @@ impl LanguageServer {
             stderr_capture,
             Some(server),
             root_path,
+            working_dir,
             code_action_kinds,
             cx,
             move |notification| {
@@ -322,6 +324,7 @@ impl LanguageServer {
         stderr_capture: Arc<Mutex<Option<String>>>,
         server: Option<Child>,
         root_path: &Path,
+        working_dir: &Path,
         code_action_kinds: Option<Vec<CodeActionKind>>,
         cx: AsyncAppContext,
         on_unhandled_notification: F,
@@ -393,6 +396,7 @@ impl LanguageServer {
             io_tasks: Mutex::new(Some((input_task, output_task))),
             output_done_rx: Mutex::new(Some(output_done_rx)),
             root_path: root_path.to_path_buf(),
+            working_dir: working_dir.to_path_buf(),
             server: Arc::new(Mutex::new(server)),
         }
     }
@@ -566,7 +570,7 @@ impl LanguageServer {
         options: Option<Value>,
         cx: &AppContext,
     ) -> Task<Result<Arc<Self>>> {
-        let root_uri = Url::from_file_path(&self.root_path).unwrap();
+        let root_uri = Url::from_file_path(&self.working_dir).unwrap();
         #[allow(deprecated)]
         let params = InitializeParams {
             process_id: None,
@@ -1171,6 +1175,7 @@ impl FakeLanguageServer {
             Arc::new(Mutex::new(None)),
             None,
             Path::new("/"),
+            Path::new("/"),
             None,
             cx.clone(),
             |_| {},
@@ -1186,6 +1191,7 @@ impl FakeLanguageServer {
                     None::<async_pipe::PipeReader>,
                     Arc::new(Mutex::new(None)),
                     None,
+                    Path::new("/"),
                     Path::new("/"),
                     None,
                     cx,

